### ISSUE TITLE
Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists.

### ISSUE THREAD

**Issue by silviudc** [rospogrigio/localtuya#1773]
## Environment

- Localtuya version:  5.2.2
- Home Assistant Core version:   2024.8.3
- [x] Does the device work using the Home Assistant Tuya Cloud component ? 
- [?] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [x] Was the device working with earlier versions of localtuya ? Which one? 
- [x] Are you using the Tuya/SmartLife App in parallel ?  Using the smartlife integration on HA but this device is disabled there. I have also tried disabling that integration and still getting the same error

## Steps to reproduce

1. Try to add new device
2. Select the curtains device
3. It populates the ids and key but when clicking add it shows the following error:
Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists.

## Provide Home Assistant traceback/logs

```
put your log output between these markers
```

## Additional information
It detects the device just fine (other devices too which do work)

![image](https://github.com/user-attachments/assets/61f45586-e579-49f0-a6fb-de3d70ce17d1)

I am seeing many of the following warnings in the HA log and they happen constantly:
![image](https://github.com/user-attachments/assets/49c30d09-683f-488f-adf9-1a99c1a2436a)

**Comment by Alessolfo10** [comment]
me too

**Comment by harjaj** [comment]
me too

**Comment by silviudc** [comment]
Reverted back to 5.2.1 and the thousands of warnings stopped but still cannot get the curtains to work

**Comment by j2g2com** [comment]
I understand that reinstalling 5.2.1 on top of it will make everything work again. It happens to me with some devices randomly.

**Comment by garry0garry** [comment]
Local Tuya 5.2.2 or 5.2.1 ...
![изображение](https://github.com/user-attachments/assets/6e43036a-d5a6-4887-90f3-07db82825546) 
![изображение](https://github.com/user-attachments/assets/6a28a1ec-f59e-4fd2-865a-e6cd487f0964)
```
This error originated from a custom integration.

Logger: custom_components.localtuya.common
Source: custom_components/localtuya/common.py:130
integration: LocalTuya (documentation, issues)
First occurred: 11:04:04 PM (114 occurrences)
Last logged: 11:08:45 PM

Missing device configuration for device_id bf68097fe7e954468daipk
Missing device configuration for device_id bf6882f2de238b3d4bqboa
```

**Comment by garry0garry** [comment]
[config_entry-localtuya-f89b78cc9fc91bc8cdf90d86e7197eba.json](https://github.com/user-attachments/files/16768504/config_entry-localtuya-f89b78cc9fc91bc8cdf90d86e7197eba.json)

**Comment by BambamNZ** [comment]
Seeing the similar error messages, but no impact as far as I have observed - granted I only have 3 devices integrate under this 

```
This error originated from a custom integration.

Logger: custom_components.localtuya.common
Source: custom_components/localtuya/common.py:130
integration: LocalTuya (documentation, issues)
First occurred: 3:41:33 PM (5060 occurrences)
Last logged: 3:54:28 PM

Missing device configuration for device_id ebe9427dc416c3379dnnkj
Missing device configuration for device_id ebca4b1972b163fdc0bara
Missing device configuration for device_id eb0c74bb125ea7d2a1u5gi
Missing device configuration for device_id 05170535d8f15b017ef8
Missing device configuration for device_id 0870171150029102609c
```

**Comment by garry0garry** [comment]
This error originated from a custom integration.

Logger: custom_components.localtuya.common
Source: custom_components/localtuya/common.py:130
integration: LocalTuya (documentation, issues)
First occurred: August 27, 2024 at 11:23:15 PM (**11588** occurrences)
Last logged: 7:26:44 AM

Missing device configuration for device_id bf6882f2de238b3d4bqboa
Missing device configuration for device_id bf68097fe7e954468daipk

**Comment by JackTalisker** [comment]
Same for me.
I noticed that this warning concerns devices that are online and configured in the Tuya app. 
It seems that does not warn about devices configured but NOT online.

**Comment by Alessolfo10** [comment]
I can't configure my Ledvance bulb even with the old version

**Comment by silviudc** [comment]
went back to 5.2.1 and now just this constantly, won't detect any of the devices:
![image](https://github.com/user-attachments/assets/13f85827-de39-4ca8-82a2-41c7b52bef84)

**Comment by flavioxavier** [comment]
Same issue here I have a light that was configured well, and stopped working, tried to remove and re-add and now I have that error, even adding the dpid manually it does't work

**Comment by ntompson** [comment]
Also getting the log spam about missing device configuration after upgrading to 5.2.2. In my case the device_id that is quoted as having missing config doesn't exist.

**Comment by schuettecarsten** [comment]
Same problem here.

**Comment by NicolasWoitchik** [comment]
Same problem here. But the device without configuration is a campaign and this type is not supported by localtuya.
Can i do something to fix that?

**Comment by niwzo** [comment]
same here as well,so gone back to using the online tuya

**Comment by MelleD** [comment]
> Same for me. I noticed that this warning concerns devices that are online and configured in the official Tuya app/integration (but not configured in Local Tuya). It seems that does not warn about other devices NOT online.

Are you sure?
I have an Eufy Cleaner which has Tuya under hood. I did not register it in der local tuya app/integration also not on the IOT platform, but the missing device warning also appears

**Comment by JackTalisker** [comment]
> Are you sure?
> I have an Eufy Cleaner which has Tuya under hood. I did not register it in der local tuya app/integration also not on the IOT platform, but the missing device warning also appears

I was telling about my devices. 
They are are configured in at least one of the two integration.

**Comment by dalto12593** [comment]
Same exact issues here.

**Comment by berkavil** [comment]
Hallo all, the same issue here. thx.

**Comment by rospogrigio** [comment]
Fixed by #1816 .
